### PR TITLE
PI-884 - Fixed End2End tests, handle multi user login for OASys, Handle pages to complete the JS and added the recent changes from Approved Premises

### PR DESCRIPTION
**Pull request by smalepati99**
The PR addresses the following:
1. Incorporates code changes related to recent "Approved Premises" changes
2. Fix for "create-contact" tests failure
3. Fix for "search-for-person.spec" tests failure
4. Fix for "create-pre-sentence-report" tests failure
5. Amended js function to add optional parameter
6. Extended time duaration for "allocate-new-person" tests
7. Fix for OASys Tests failures due Concurrent Login issues(tests will now have distinct logins)
8. Added code to handle WIP - Work In Progress Assessments
9. Fix for OASys test are passing in the local machine and failing in the CI pipeline

### CODE REVIEW

File: steps/delius/contact/create-contact.ts [hunk 1]
```diff
@@ -1,22 +1,26 @@
 import { expect, type Page } from '@playwright/test'
 import { findContactsByCRN } from './find-contacts.js'
 import { fillDate, fillTime, selectOptionAndWait } from '../utils/inputs.js'
-import { Contact, data, Team } from '../../../test-data/test-data.js'
+import {Contact, data, Staff, Team} from '../../../test-data/test-data.js'
 import { doUntil } from '../utils/refresh.js'
 
-export const createContact = async (page: Page, crn: string, options: Contact) => {
+export const createContact = async (page: Page, crn: string, options: { date: Date; allocation: { staff: Staff; team: Team; providerName: string }; category: string; type: string; relatesTo: string }) => {
```
Comment (marcus-bcl) [inline steps/delius/contact/create-contact.ts]: I don't think you need this change - it looks like it's just redefining the Contact class?

File: tests/workforce-allocations-to-delius/allocate-new-person.spec.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ const practitioner: Allocation = { staff: data.staff.allocationsTester1, team: d
 const anotherPractitioner: Allocation = { staff: data.staff.allocationsTester2, team: data.teams.allocationsTestTeam }
 
 test('Allocate new person', async ({ page }) => {
+    test.slow()
```
Comment (marcus-bcl) [inline tests/workforce-allocations-to-delius/allocate-new-person.spec.ts]: Are these tests taking longer than 3 minutes now?

**Review by marcus-bcl — COMMENTED** [review]
Looks like the code formatter hasn't kicked in, possibly due to the merge conflicts.

File: steps/delius/contact/create-contact.ts (hunk 1 above)
Comment (smalepati99) [reply steps/delius/contact/create-contact.ts]: Reverted it back

File: tests/workforce-allocations-to-delius/allocate-new-person.spec.ts (hunk 2 above)
Comment (smalepati99) [reply tests/workforce-allocations-to-delius/allocate-new-person.spec.ts]: Reverted back

**Review by anthony-britton-moj — APPROVED** [review]

**Pull request merged**